### ISSUE TITLE
Allow facts be reloaded dynamically during a deploy

### ISSUE THREAD

**Issue by ghost** [Fizzadar/pyinfra#781]
## Is your feature request related to a problem? Please describe
When trying to deploy a server I need to wait until a container has finished what it is doing (in this case pushing container images to a registry). I know when this is done by checking if the container exits in a running state. I can only check this one time at the beginning of a deploy using:
```py
host.get_fact(Command,
              'sudo runuser -l -s /bin/sh tinkerbell -c "podman ps --noheading -f name=images-to-local-registry"',
)
```

Which, maybe I am doing this wrong, but I also cannot inject variables, or environment variables into this fact, (so I cannot switch user or sudo in the fact, so I sudo in the command). That is an aside I guess. The main issue is that I want to run this in a loop so that I can tell when container no longer exists. It appears that this is not possible, at least, I have not been able to see how. Possibly this is not a good idea for some reason, or, maybe this should be written into an operation. Either way, I think that you should be able to dynamically reload facts/specific facts at any point during a deploy.

## Describe the solution you'd like
Allow facts, or at least one particular fact, to be reloaded anytime during a deploy.  When trying to assess the state of something, say the status of a systemd unit, I want to be able re-assess the state of the systemd unit  in a loop so that I can make sure that an action has completed properly before continuing on with the deploy.

cc: @jmpolom @storrgie

**Comment by Fizzadar** [comment]
Fantastic idea! This could be a simple `host.reload_fact(...)` that works like `get_fact` but ignores any cached value.

**Comment by Fizzadar** [comment]
This has been released in `v2.1`.
